### ISSUE TITLE
running Genie on Ubuntu with Apache2

### ISSUE THREAD

**Issue by Pipsy100** [GenieFramework/Genie.jl#372]
I want to run Genie on Apache2 as a webservice and I am looking for documentation to create the .conf file in sites-available. Is there documentation about this implementation

**Comment by essenciary** [comment]
@Pipsy100 given that Apache does not work with Julia in the same way it does with say PHP or Ruby, the only approach is to run Apache as a web proxy for Genie apps. That is, set up your Apache host as you'd normally do and proxy all the traffic for the host to the Genie app (usually running on port 0.0.0.0:8000). The Genie app itself should be demonized so that it's automatically restarted in case it crashes. 

If you need more details about how to do the above I can share some of the setup we use at work - and add a new section to the docs next week.

**Comment by Pipsy100** [comment]
Adrian,

I am trying to  get in running on Nginx in 《domainname》/genie.

I cannot get it working with the instructions from the website. Maybe you can help me out.

Thx a lot.

23 May 2021 14:49:38 Adrian Salceanu ***@***.***>:

**Comment by essenciary** [comment]
@Pipsy100 Yes, the default setup expects for the app to run as `genie.domainname/`. But there is support to run it in a folder as well, though it's not documented yet (I used it to run Genie apps on JuliaHub). For far did you get with the setup? What is the current situation (what works and what not)?

**Comment by Pipsy100** [comment]
Hello Adrian, thx for helping me out in the first place. I am new to Nginx and Genie.

I have a Nginx server running on an Ubuntu VPS with https://kentekenkaartuitlezen.nl/

See here below my default configuration in sites-enabled.

I am able to start Genie locally:  Web Server starting at http://0.0.0.0:8000 -

the application directory on the VPS is /var/www/kentekenkaartuitlezen/genie


Now I would like to run Genie under https://kentekenkaartuitlezen.nl/genie
And start it preferably from a separate nginx config file.

I follow these instruction: https://genieframework.github.io/Genie.jl/dev/tutorials/92--Deploying_Genie_Server_Apps_with_Nginx.html
But i cannot get it running.

Thx a lot again in advance again.

Pips


server {
root /var/www/kentekenkaartuitlezen;

index index.html index.htm index.nginx-debian.html index.php;

server_name kentekenkaartuitlezen.nl;

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/kentekenkaartuitlezen.nl-0001/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/kentekenkaartuitlezen.nl-0001/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}

server {
    if ($host = kentekenkaartuitlezen.nl) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


listen 80 default_server;
listen [::]:80 default_server;

server_name kentekenkaartuitlezen.nl;
    return 404; # managed by Certbot


}

________________________________
